tests/unit/ReportTitleUtilsTest.ts

```diff
@@ -0,0 +1,216 @@
+
+            expect(result).toHaveLength(1);
+            expect(result.at(0)?.value).toEqual({
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                expensify_text_title: '',
+            });
+        });
+
+        it('should handle special characters in title field', () => {
+            const specialTitle = 'Title with "quotes" & <tags>';
+
+            const result = updateTitleFieldWithExactValue(mockReportID, specialTitle);
+
+            expect(result.at(0)?.value).toEqual({
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                expensify_text_title: specialTitle,
+            });
+        });
+    });
+
+    describe('removeTitleFieldFromReport', () => {
+        const mockReportID = '12345';
+
+        it('should return optimistic update with null value for valid reportID', () => {
+            const result = removeTitleFieldFromReport(mockReportID);
+
+            expect(result).toHaveLength(1);
+            expect(result.at(0)).toEqual({
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${mockReportID}`,
+                value: {
+                    // eslint-disable-next-line @typescript-eslint/naming-convention
+                    expensify_text_title: null,
+                },
+            });
+        });
+    });
+
+    describe('shouldUpdateTitleField', () => {
+        const mockReportID = '12345';
+        const mockTitleField = {
+            defaultValue: 'Test Title',
+        };
+
+        beforeEach(() => {
+            mockedReportUtils.getReportNameValuePairs.mockReturnValue({
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                expensify_text_title: mockTitleField,
+            });
+        });
+
+        it('should return false when report is chat report', () => {
+            const mockReport: Report = {
+                reportID: mockReportID,
+                type: CONST.REPORT.TYPE.CHAT,
+            } as Report;
+
+            mockedReportUtils.isChatReport.mockReturnValue(true);
+
+            const result = shouldUpdateTitleField(mockReport);
+
+            expect(mockedReportUtils.isChatReport).toHaveBeenCalledWith(mockReport);
+            expect(result).toBe(false);
+        });
+
+        it('should return false when report has no title field in RNVP', () => {
+            const mockReport: Report = {
+                reportID: mockReportID,
+                type: CONST.REPORT.TYPE.EXPENSE,
+            } as Report;
+
+            mockedReportUtils.isChatReport.mockReturnValue(false);
+            mockedReportUtils.getReportNameValuePairs.mockReturnValue({});
+
+            const result = shouldUpdateTitleField(mockReport);
+
+            expect(result).toBe(false);
+        });
+
+        it('should return true when non-chat report has title field', () => {
```

Comment:
Am I missing something or shouldn't we have to mock that the report name value pairs returns the title field?